### ISSUE TITLE
Theme check action fails if no lintable files were changed in a pull request

### ISSUE THREAD

**Issue by MauriceArikoglu** [Shopify/theme-check-action#28]
https://github.com/Shopify/theme-check-action/blob/fc581c95120e0db82b86ae6fa8c507cba7648aba/entrypoint.sh#LL62C1-L63C8

If no lintable files are found the exit code from the theme-check call causes the action to fail, even though no errors and no warnings were found.

**Comment by snacsnoc** [comment]
@MauriceArikoglu I was able to debug this a bit more and went down a rabbit hole. It seems theme-check returns a few 'offences' when no lintable files are found. Since the theme-check action runs on changed files (see https://github.com/Shopify/theme-check-action/blob/main/entrypoint.sh#L67), the default theme-check categories attempt to run. The changed files are saved to `/tmp/diff.log and then the files are added to the `themePath` which theme-check will run on (see https://github.com/Shopify/theme-check-action/blob/main/index.js#L37). So, getting to the actual issue, theme-check returns:
```
No theme files found.
```
which results in exit code 1, indicating a fail even though that's not necessarily true (in some cases).
theme-check will exit when no matching file/directory patterns are found (see https://github.com/Shopify/theme-check/blob/main/lib/theme_check/cli.rb#L195), which is defined here: https://github.com/Shopify/theme-check/blob/main/lib/theme_check/theme.rb#L55

So in my case, I'm updating a `.yml` file and theme-check is failing because there are no JSON, liquid files and the `assets/ dir` doesn't exist. This all makes sense, and defining a solution for non-lintable files is a bit more straightforward now.

I was able to 'solve' this by capturing the output of theme-check and checking for `No theme files found.` and returning 0 as the `code`.

Example (from https://github.com/Shopify/theme-check-action/blob/81e511eee8b49165060e91a685441176c079f116/entrypoint.sh#L62):
```
# Capture both stdout and stderr from theme-check
output=$(run theme-check $flags -o json "$theme_root" 2>&1)
code=$?

# Check if 'No theme files found.' is in the output
if echo "$output" | grep -q "No theme files found."; then
    echo "No theme files found, but proceeding without error."
    code=0
fi

# Output the results to a file
echo "$output" > /tmp/results.json
```
It's not the most elegant solution, but this resolves the issue. Open to discussion on this, thanks!

**Comment by MauriceArikoglu** [comment]
@snacsnoc Thank you Easton, for digging into this and providing a workaround. I came to a similar conclusion with the changed files, but did not dig deep enough to find a suitable workaround.

Would be nice to add this to a caveats section or refactor the action appropriately.

Have a good day and best wishes for your 2024.
